src/components/CollaboratorPage/index.jsx

```diff
@@ -0,0 +1,165 @@
+  constructor(props) {
+    super(props);
+    this.handleCollaboratorCreate = this.handleCollaboratorCreate.bind(this);
+    this.handleCollaboratorEdit = this.handleCollaboratorEdit.bind(this);
+  }
+
+  handleCollaboratorCreate(fieldValues) {
+    const {
+      createCollaborator,
+      sourceInfo: { referrer },
+    } = this.props;
+    createCollaborator(fieldValues, referrer);
+  }
+
+  handleCollaboratorEdit(fieldValues) {
+    const {
+      editCollaborator,
+      sourceInfo: { referrer },
+      location: {
+        state: {
+          uuid,
+        },
+      },
+    } = this.props;
+
+    const collaboratorData = { ...fieldValues, uuid };
+    if (!collaboratorData.image.startsWith('data:')) {
+      // Only send image if a new one is being uploaded
+      delete collaboratorData.image;
+    }
+    editCollaborator(collaboratorData, referrer);
+  }
+
+  render() {
+    const {
+      sourceInfo,
+      location,
+      collaboratorInfo,
+    } = this.props;
+
+    const isCreateForm = !this.props.editCollaborator;
+
+    if (!isCreateForm && (!location.state || !location.state.uuid)) {
+      return (
+        <StatusAlert
+          id="error"
+          alertType="danger"
+          title="Could not load page: "
+          message="Direct access to collaborators not supported"
+        />
+      );
+    }
+
+    const titleText = isCreateForm ? 'Create New Collaborator' : 'Edit Collaborator';
+    const handleSubmit = (isCreateForm
+      ? this.handleCollaboratorCreate
+      : this.handleCollaboratorEdit);
+
+    const { isSaving = false, error: collabError } = collaboratorInfo;
+
+    const errorArray = [];
+
+    if (collabError) {
+      collabError.forEach((error, index) => {
+        errorArray.push(error);
+        if (index < collabError.length) {
+          errorArray.push(<br />);
+        }
+      });
+    }
+    const { referrer } = sourceInfo;
+
+    return (
+      <>
+        <Helmet>
+          <title>{titleText}</title>
+        </Helmet>
+        <PageContainer>
+          { referrer
+          && (
+            <StatusAlert
+              id="sent-from-edit-course-info"
+              alertType="info"
+              message="The data you entered on the course edit screen is saved. You will return to that page when you have finished updating collaborator information."
+              dismissible
+            />
+          )}
+          <div>
+            <h2>{titleText}</h2>
+            <hr />
+            <CollaboratorForm
+              id="create-collaborator-form"
+              onSubmit={handleSubmit}
+              isSaving={isSaving}
+              isCreateForm={isCreateForm}
+              initialValues={{
+                name: (location && location.state) ? location.state.name : null,
+                image: (location && location.state) ? location.state.imageUrl : null,
+              }}
+              {...this.props}
+            />
+            { errorArray.length > 1 && (
```

Comment:
Should this be `> 0`?

Reply:
tbh I copied and pasted this from the StafferPage. I'll investigate it a little more

Reply:
looks like it should indeed be > 0. not sure what's up with the StafferPage one. updated.